pkgs/development/compilers/dmd/default.nix

```diff
@@ -109,98 +114,118 @@ stdenv.mkDerivation rec {
     rm dmd/test/runnable/gdb15729.sh
     rm dmd/test/runnable/gdb4149.d
     rm dmd/test/runnable/gdb4181.d
+  ''
 
-    # Grep'd string changed with gdb 12
+  # Grep'd string changed with gdb 12
+  + ''
     substituteInPlace druntime/test/exceptions/Makefile \
       --replace 'in D main (' 'in _Dmain ('
   ''
 
+  # We're using gnused on all platforms
+  + ''
```

Comment:
Please revert this.

Reply:
Please elaborate why. Not patching this breaks the tests on Darwin because it detects OSX and tries to use bsdsed-style flag formatting, which makes the gnused in the stdenv reject the command.

Reply:
See #163989.

Reply:
I meant splitting out the comments into nix code and then concatenating strings. I didn't even see that you changed something other than style in here.

Reply:
You're asking for more than a revert then because the rest of the phase already handled comments like that, this change was just for consistency.

Reply:
I commented what I mean if that wasn't clear.

Reply:
I've restructured the `patchPhase` to move such comments into the shell code and shuffled around the blocks so unconditional & conditional code isn't needlessly interspersed. I hope this is better now.